### PR TITLE
Fix pagination appeal type sort order

### PR DESCRIPTION
**Pull request by hschallhorn**
Resolves bug where paginated queues did not bubble aod cases to the top

### Description
Fixed task sorting on the back end to always sort aod to the top when sorting by appeal type.

### Acceptance Criteria
- [ ] When pagination is enabled, the default sort order on page load bubbles AOD cases to the top

### Testing Plan
1. Ensure user queue pagination is enabled
    ```ruby
    FeatureToggle.enable!(:user_queue_pagination, users: [User.find_by(css_id: "CSS_ID1").css_id])
    ```
1. On master, log in as CSS_ID1 and go to their queue.
1. Notice the AOD cases are not listed first
1. pull this branch and reload
1. Notice AOD cases listed first

### User Facing Changes

 BEFORE|AFTER
 ---|---
Screen Shot 2020-05-28 at 11 54 06 AM|Screen Shot 2020-05-28 at 12 00 23 PM

### CODE REVIEW

File: app/models/task_sorter.rb [hunk 1]
```diff
@@ -71,6 +73,12 @@ def task_type_order_clause
     "position(#{task_type_sort_position}) #{sort_order}"
   end
 
+  def appeal_type_order_clause
+    "cached_appeal_attributes.is_aod DESC, "\
+    "cached_appeal_attributes.case_type #{sort_order}, "\
+    "cached_appeal_attributes.docket_number #{sort_order}"
+  end
```
Comment (hschallhorn) [inline app/models/task_sorter.rb:80]: Previous sorting logic: https://github.com/department-of-veterans-affairs/caseflow/pull/11750/files#diff-9e90b6f8519fd9c2b8d7d1b032fe4f11L69

File: spec/factories/appeal.rb [hunk 2]
```diff
@@ -120,7 +120,7 @@
         another_claimant = create(:claimant)
         create(:advance_on_docket_motion, person: claimant.person, granted: true)
         create(:advance_on_docket_motion, person: another_claimant.person, granted: false)
-        [claimant, another_claimant]
+        [another_claimant, claimant]
```
Comment (hschallhorn) [inline spec/factories/appeal.rb:123]: The claimant for an ama appeal is the[ last claimant](https://github.com/department-of-veterans-affairs/caseflow/blob/566b4e93f8cf213bddd75ae6065e4ce050469a62/app/models/decision_review.rb#L152-L153). This swap ensures the claimant for the case is the one with the granted AOD motion. The test below wasn't actually creating any aod cases, which is why it was not failing.

File: spec/models/task_sorter_spec.rb [hunk 3]
```diff
@@ -316,8 +316,9 @@
 
         it "sorts by AOD status, case type, and docket number" do
           expected_order = CachedAppeal.all.sort_by do |cached_appeal|
-            [cached_appeal.is_aod ? 1 : 0, cached_appeal.case_type, cached_appeal.docket_number]
+            [cached_appeal.is_aod ? 0 : 1, cached_appeal.case_type, cached_appeal.docket_number]
```
Comment (hschallhorn) [inline spec/models/task_sorter_spec.rb:319]: [Revert bad change](https://github.com/department-of-veterans-affairs/caseflow/pull/11750/files#diff-327a8496405fce8f7b05efc925e95f07R326). We want to sort is_aod = true first (0 < 1)

File: spec/models/task_sorter_spec.rb [hunk 4]
```diff
@@ -316,8 +316,9 @@
...
           end
+          expect(expected_order.first.is_aod).to eq true
```
Comment (hschallhorn) [inline spec/models/task_sorter_spec.rb:321]: Ensures we [actually created an aod appeal](https://github.com/department-of-veterans-affairs/caseflow/pull/14419/files#diff-12e1d20563b05c69deedbaef626c0aceR123)

File: spec/factories/appeal.rb (hunk 2 above)
Comment (hschallhorn) [reply spec/factories/appeal.rb:123]: https://github.com/department-of-veterans-affairs/caseflow/blob/0ad09ed61c556d45709d50e07a871c5fbf363578/spec/models/task_sorter_spec.rb#L306

File: spec/models/task_pager_spec.rb [hunk 5]
```diff
@@ -407,7 +407,7 @@
 
       it "sorts by AOD status, case type, and docket number" do
         expected_order = CachedAppeal.all.sort_by do |cached_appeal|
-          [cached_appeal.is_aod ? 1 : 0, cached_appeal.case_type, cached_appeal.docket_number]
+          [cached_appeal.is_aod ? 0 : 1, cached_appeal.case_type, cached_appeal.docket_number]
```
Comment (hschallhorn) [inline spec/models/task_pager_spec.rb:410]: Same as [below](https://github.com/department-of-veterans-affairs/caseflow/pull/14419/commits/0ad09ed61c556d45709d50e07a871c5fbf363578#r431964031)!

File: spec/models/task_sorter_spec.rb (hunk 3 above)
Comment (lomky) [reply spec/models/task_sorter_spec.rb:319]: I believe you, but I honestly cannot parse this line for the life of me. Is there some way we can clarify it either via a named method or a comment?

File: spec/models/task_sorter_spec.rb (hunk 4 above)
Comment (lomky) [reply spec/models/task_sorter_spec.rb:321]: great coverage check

File: spec/factories/appeal.rb (hunk 2 above)
Comment (lomky) [reply spec/factories/appeal.rb:123]: I think this comment should go in the code!

**Pull request merged**